### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by saketkum11**
- api call
- getvideo
- watchLater page

### CODE REVIEW

File: src/Component/Card/Card.jsx [hunk 1]
```diff
@@ -1,72 +1,103 @@
...
 import { BsThreeDotsVertical } from "react-icons/bs";
+import { MdPlaylistAdd, MdRemoveCircle } from "react-icons/md";
 import { NavLink } from "react-router-dom";
-import { useVideo } from "../../Context/Video-Context/video-context";
-import { Navigation } from "../Nav/Navbar";
 import "./Card.css";
-const Card = () => {
-  const { videoData } = useVideo();
+import { useWatch } from "../../Context/WatchLater-context/Watch-context";
+import { useState } from "react";
```
Comment (Sharmaryan) [inline src/Component/Card/Card.jsx:7]: use absolute path instead of ../../ this

File: src/Context/Auth-context/Auth-context.js [hunk 2]
```diff
@@ -24,15 +23,23 @@ const AuthProvider = ({ children }) => {
       console.error(error);
     }
   };
+
   const loginHandler = async ({ email, password }) => {
-    if (isAuth) {
+    console.log("from login handler");
+    console.log("isauth", isAuth);
+    if (!isAuth) {
```
Comment (Sharmaryan) [inline src/Context/Auth-context/Auth-context.js:28]: remove console log

File: src/Context/Video-Context/video-context.js [hunk 3]
```diff
@@ -12,13 +12,14 @@ const VideoProvider = ({ children }) => {
       try {
         const response = await axios.get("/api/videos");
         setVideoData(response.data.videos);
+        console.log("response from video context", response);
       } catch (error) {
```
Comment (Sharmaryan) [inline src/Context/Video-Context/video-context.js:16]: also here

File: src/Context/WatchLater-context/Watch-context.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            authorization: tokenData,
+          },
+        });
+        setWatchVideo(response.data.watchlater);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    getWatchLaterData();
+  }, []);
```
Comment (Sharmaryan) [inline src/Context/WatchLater-context/Watch-context.js:27]: it is a good practice to use IIFE here

File: src/Pages/VideoPlayList/VideoList.jsx [hunk 5]
```diff
@@ -3,11 +3,31 @@ import { Card, SideNav } from "../index";
 import "./VideoList.css";
 const VideoList = () => {
   const { videoData } = useVideo();
+  console.log("videiData from ", videoData);
   return (
```
Comment (Sharmaryan) [inline src/Pages/VideoPlayList/VideoList.jsx:6]: remove this also

File: src/Pages/Watch-later/WatchLater.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { useVideo } from "../../Context/Video-Context/video-context";
+import { useWatch } from "../../Context/WatchLater-context/Watch-context";
+import { SideNav, Card } from "../index";
+const WatchLater = () => {
```
Comment (Sharmaryan) [inline src/Pages/Watch-later/WatchLater.jsx:5]: here also you can use absolute path

**Review by Sharmaryan — APPROVED** [review]

File: src/Component/Card/Card.jsx (hunk 1 above)
Comment (saketkum11) [reply src/Component/Card/Card.jsx:7]: this something new for me.

File: src/Context/Auth-context/Auth-context.js (hunk 2 above)
Comment (saketkum11) [reply src/Context/Auth-context/Auth-context.js:28]: ok I have done changes

File: src/Context/Video-Context/video-context.js (hunk 3 above)
Comment (saketkum11) [reply src/Context/Video-Context/video-context.js:16]: I have done changes

File: src/Context/WatchLater-context/Watch-context.js (hunk 4 above)
Comment (saketkum11) [reply src/Context/WatchLater-context/Watch-context.js:27]: okk .

File: src/Pages/VideoPlayList/VideoList.jsx (hunk 5 above)
Comment (saketkum11) [reply src/Pages/VideoPlayList/VideoList.jsx:6]: I have done the changes

File: src/Component/Card/Card.jsx (hunk 1 above)
Comment (Sharmaryan) [reply src/Component/Card/Card.jsx:7]: I will help with this

**Review by shubhamkumar648 — APPROVED** [review]
Good use toggle functionality, as i Aryan already mentions require changes,remaining code  is good  and understandable, good to go

**Pull request merged**